### PR TITLE
fix interaction of space flag and '+' flag, as well as '-' flag and '…

### PR DESCRIPTION
**Pull request by rimathia**
…0' flag

I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

File: include/fmt/printf.h [hunk 1]
```diff
@@ -405,7 +405,9 @@ void basic_printf_context<OutputIt, Char>::parse_flags(format_specs& specs,
       specs.fill[0] = '0';
       break;
     case ' ':
-      specs.sign = sign::space;
+      if (specs.sign != sign::plus) {
```
Comment (rimathia) [inline include/fmt/printf.h:408]: from the opengroup description: "This means that if the <space> and '+' flags both appear, the <space> flag shall be ignored."

File: include/fmt/printf.h [hunk 2]
```diff
@@ -517,10 +519,11 @@ OutputIt basic_printf_context<OutputIt, Char>::format() {
     if (specs.alt && visit_format_arg(detail::is_zero_int(), arg))
       specs.alt = false;
     if (specs.fill[0] == '0') {
-      if (arg.is_arithmetic())
+      if (arg.is_arithmetic() && specs.align != align::left)
         specs.align = align::numeric;
       else
-        specs.fill[0] = ' ';  // Ignore '0' flag for non-numeric types.
+        specs.fill[0] = ' ';  // Ignore '0' flag for non-numeric types or if '-'
+                              // flag is also present.
```
Comment (rimathia) [inline include/fmt/printf.h:526]: opengroup description: "If the `0` and `-` flags both appear, the `0` flag is ignored."

**Comment by rimathia** [comment]
Hi, since the behaviour changes implemented here seem pretty clearly in line with the opengroup description of the printf familiy I have skipped opening an issue.

**Comment by vitaut** [comment]
This is great, thanks!

**Pull request merged**